recipes/caf/all/conanfile.py

```diff
@@ -0,0 +1,140 @@
+                self._cmake.definitions["CAF_BUILD_STATIC"] = self._is_static
+                self._cmake.definitions["CAF_BUILD_STATIC_ONLY"] = self._is_static
+            else:
+                self._cmake.definitions["CMAKE_CXX_STANDARD"] = "17"
+                self._cmake.definitions["CAF_ENABLE_OPENSSL_MODULE"] = self._has_openssl
+                for define in ["CAF_ENABLE_EXAMPLES", "CAF_ENABLE_TOOLS", "CAF_ENABLE_TESTING"]:
+                    self._cmake.definitions[define] = "OFF"
+            if tools.os_info.is_macos and self.settings.arch == "x86":
+                self._cmake.definitions["CMAKE_OSX_ARCHITECTURES"] = "i386"
+            self._cmake.definitions["CAF_LOG_LEVEL"] = self.options.log_level
+            if self.settings.os == 'Windows':
+                self._cmake.definitions["OPENSSL_USE_STATIC_LIBS"] = True
+                self._cmake.definitions["OPENSSL_MSVC_STATIC_RT"] = True
+            elif self.settings.compiler == 'clang':
+                self._cmake.definitions["PTHREAD_LIBRARIES"] = "-pthread -ldl"
+            else:
+                self._cmake.definitions["PTHREAD_LIBRARIES"] = "-pthread"
+                if self.settings.compiler == "gcc" and Version(self.settings.compiler.version.value) < "5.0":
+                    self._cmake.definitions["CMAKE_SHARED_LINKER_FLAGS"] = "-pthread"
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        if "patches" in self.conan_data and self.version in self.conan_data["patches"]:
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
+        cmake = self._cmake_configure()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE*", dst="licenses", src=self._source_subfolder)
+        cmake = self._cmake_configure()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        suffix = "_static" if self._is_static and self.version == "0.17.6" else ""
+
+        self.cpp_info.names["cmake_find_package"] = "CAF"
+        self.cpp_info.names["cmake_find_package_multi"] = "CAF"
+
+        self.cpp_info.components["core"].names["pkg_config"] = f"caf_core{suffix}"
+        self.cpp_info.components["core"].libs = [f"caf_core{suffix}"]
+        if self.settings.os == "Linux":
+            self.cpp_info.components["core"].system_libs = ["pthread", "m"]
+
+        self.cpp_info.components["io"].names["pkg_config"] = f"caf_io{suffix}"
+        self.cpp_info.components["io"].libs = [f"caf_io{suffix}"]
+        self.cpp_info.components["io"].requires = ["core"]
+        if self.settings.os == "Windows":
+            self.cpp_info.components["io"].system_libs = ["ws2_32", "iphlpapi", "psapi"]
+
+        if self._has_openssl:
+            self.cpp_info.components["openssl"].names["pkg_config"] = f"caf_openssl{suffix}"
```

Comment:
```suggestion
            self.cpp_info.components["openssl"].names["pkg_config"] = "caf_openssl{}".format(suffix)
```

Cannot use f-string because it requires python 3.6